experimental/framework/fluid-static/src/signalManager.ts

```diff
@@ -0,0 +1,177 @@
+
+    private readonly managerId: string | undefined;
+
+    constructor(
+        /**
+         * Object to wrap that can submit and listen to signals
+         */
+        private readonly signaler: ISignaler,
+        /**
+         * Optional id to assign to this manager that will be attached to
+         * signal names.  Useful to avoid collisions if there are multiple
+         * signal users at the Container level
+         */
+        managerId?: string,
+    ) {
+        super();
+        this.managerId = managerId ? `#${managerId}` : undefined;
+        this.signaler.on("signal", (message: IInboundSignalMessage, local: boolean) => {
+            const listeners = this.listenerMap.get(message.type);
+            const clientId = message.clientId;
+            // Only call listeners when the runtime is connected and if the signal has an
+            // identifiable sender clientId.  The listener is responsible for deciding how
+            // it wants to handle local/remote signals
+            // eslint-disable-next-line no-null/no-null
+            if (listeners !== undefined && this.signaler.connected && clientId !== null) {
+                listeners.forEach((listener) => {
+                    listener(clientId, local, message.content);
+                });
+            }
+        });
+    }
+
+    private getManagerSignalName(signalName: string): string {
+        return this.managerId ? `${signalName}${this.managerId}` : signalName;
+    }
+
+    private getBroadcastSignalName(signalName: string): string {
+        return `${signalName}#req`;
+    }
+
+    // ISignalManager methods
+
+    public registerListener(
+        signalName: string,
+        listener: SignalListener,
+    ): ISignalManager {
+        const managerSignalName = this.getManagerSignalName(signalName);
+        let listenerList = this.listenerMap.get(managerSignalName);
+        if (listenerList === undefined) {
+            listenerList = [];
+            this.listenerMap.set(managerSignalName, listenerList);
+        }
+        listenerList.push(listener);
+        return this;
+    }
+
+    public deregisterListener(
+        signalName: string,
+        listener: SignalListener,
+    ): ISignalManager {
+        const managerSignalName = this.getManagerSignalName(signalName);
+        const listenerList = this.listenerMap.get(managerSignalName);
+        if (listenerList !== undefined) {
+            const index = listenerList.lastIndexOf(listener);
+            if (index !== -1) {
+                listenerList.splice(index, 1);
+            }
+        }
+        return this;
+    }
+
+    public submitSignal(
+        signalName: string,
+        payload?: Jsonable,
+    ) {
+        const managerSignalName = this.getManagerSignalName(signalName);
+        if (this.signaler.connected) {
+            this.signaler.submitSignal(managerSignalName, payload);
+        }
+    }
+
+    public registerBroadcastListener(
```

Comment:
What does "broadcast" signify here distinct from a non-broadcast signal?  It looks like to a caller these would be pretty equivalent, do we need both?

Reply:
No technical difference, this is mostly to be more explicit about the pattern for requesting a data broadcast that often gets used.  I think it's worth keeping because it lets the user think of things as two operations on the same signal, rather than two distinct signals (even if that's what it is under the hood)